geth/jail/internal/timers/timers_test.go

```diff
@@ -0,0 +1,140 @@
+package timers_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/robertkrimen/otto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/status-im/status-go/geth/jail/internal/loop"
+	"github.com/status-im/status-go/geth/jail/internal/timers"
+	"github.com/status-im/status-go/geth/jail/internal/vm"
+)
+
+func TestSetTimeout(t *testing.T) {
+	v, l := newVM()
+
+	err := timers.Define(v, l)
+	require.NoError(t, err)
+
+	ch := make(chan struct{})
+	err = v.Set("__capture", func() {
+		defer func() { ch <- struct{}{} }()
+	})
+	require.NoError(t, err)
+
+	err = l.Eval(`setTimeout(function(n) {
+		if (Date.now() - n < 50) {
+			throw new Error('timeout was called too soon');
+		}
+		__capture();
+	}, 50, Date.now());`)
+	require.NoError(t, err)
+
+	select {
+	case <-ch:
+	case <-time.After(1 * time.Second):
+		require.Fail(t, "test timed out")
+		return
+	}
+}
+
+func TestClearTimeout(t *testing.T) {
+	v, l := newVM()
+
+	err := timers.Define(v, l)
+	require.NoError(t, err)
+
+	ch := make(chan struct{})
+	err = v.Set("__shouldNeverRun", func() {
+		defer func() { ch <- struct{}{} }()
+	})
+	require.NoError(t, err)
+
+	err = l.Eval(`clearTimeout(setTimeout(function() {
+		__shouldNeverRun();
+	}, 50));`)
+	require.NoError(t, err)
+
+	select {
+	case <-ch:
+		require.Fail(t, "should never run")
+	case <-time.After(100 * time.Millisecond):
+	}
+}
+
+func TestSetInterval(t *testing.T) {
+	v, l := newVM()
+
+	err := timers.Define(v, l)
+	require.NoError(t, err)
+
+	ch := make(chan struct{})
+	err = v.Set("__done", func() {
+		defer func() { ch <- struct{}{} }()
+	})
+	require.NoError(t, err)
+
+	err = l.Eval(`
+		var c = 0;
+		var iv = setInterval(function() {
+			if (c === 1) {
+				clearInterval(iv);
+				__done();
+			}
+			c++;
+		}, 50);
+	`)
+	require.NoError(t, err)
+
+	select {
+	case <-ch:
+		value, err := v.Get("c")
+		require.NoError(t, err)
+		n, err := value.ToInteger()
+		require.NoError(t, err)
+		require.Equal(t, 2, int(n))
+	case <-time.After(1 * time.Second):
+		require.Fail(t, "test timed out")
+	}
+}
+
+func TestClearIntervalImmediately(t *testing.T) {
+	v, l := newVM()
+
+	err := timers.Define(v, l)
+	require.NoError(t, err)
+
+	ch := make(chan struct{})
+	err = v.Set("__shouldNeverRun", func() {
+		defer func() { ch <- struct{}{} }()
```

Comment:
Maybe, just fail it here using `require`? Generally, I see room for improvement for most tests readability.

Reply:
Done! 👍